csrc/device_lower/pass/predicate.cpp

```diff
@@ -209,6 +209,27 @@ class ConditionalFromPredicateModifier : public kir::ExprMutator {
         // here.
         return IrBuilder::create<Val>(true, DataType::Bool);
       }
+      case PredicateType::ElectSync: {
```

Comment:
Do we need a separate `PredicateType::ElectSync` predicate type? Should we just use whatever the original predicate type it has, and if the conditional happen to be `tidx == 0 && tidy == 0 && tidz == 0`, we convert it to the `elec_sync()` conditional? What do you think? @naoyam

Reply:
FYI, I separated the `ElectSync` PredicateType changes into https://github.com/NVIDIA/Fuser/pull/2923.